pkg/virt-controller/services/rendervolumes.go

```diff
@@ -332,57 +332,95 @@ func withAccessCredentials(accessCredentials []v1.AccessCredential) VolumeRender
 	}
 }
 
-func withTPM(vmi *v1.VirtualMachineInstance) VolumeRendererOption {
+func PathForSwtpm(vmi *v1.VirtualMachineInstance) string {
+	swtpmPath := "/var/lib/libvirt/swtpm"
+	if util.IsNonRootVMI(vmi) {
+		swtpmPath = filepath.Join(util.VirtPrivateDir, "libvirt", "qemu", "swtpm")
+	}
+
+	return swtpmPath
+}
+
+func PathForSwtpmLocalca(vmi *v1.VirtualMachineInstance) string {
+	localCaPath := "/var/lib/swtpm-localca"
+	if util.IsNonRootVMI(vmi) {
+		localCaPath = filepath.Join(util.VirtPrivateDir, "var", "lib", "swtpm-localca")
+	}
+
+	return localCaPath
+}
+
+func PathForNVram(vmi *v1.VirtualMachineInstance) string {
+	nvramPath := "/var/lib/libvirt/swtpm"
+	if util.IsNonRootVMI(vmi) {
+		nvramPath = filepath.Join(util.VirtPrivateDir, "libvirt", "qemu", "nvram")
+	}
+
+	return nvramPath
+}
+
+func withVMState(vmi *v1.VirtualMachineInstance) VolumeRendererOption {
```

Comment:
What about calling this `withBackendStorage()`? VMState sounds a little bit vague(?)